src/plugins/intel_gpu/src/graph/network.cpp

```diff
@@ -1508,4 +1516,57 @@ void network::assign_variables_memories(variables_states_map &&variables_memorie
     }
 }
 
+void network::assign_variables_memories() {
+    for (auto primitive : _variable_state_primitives) {
+        if (const auto& memory_state_primitive = std::dynamic_pointer_cast<memory_state::variable>(primitive)) {
+            auto it = _variables_states.find(memory_state_primitive->variable_id());
+            if (it != _variables_states.end()) {
+                primitive->set_output_memory(it->second->memory, false);
+            }
+        }
+    }
+}
+
+void network::update_variable_memory(const std::string& variable_id, const cldnn::layout& layout) {
+    auto it = _variables_states.find(variable_id);
+    if (it != _variables_states.end()) {
+        if (it->second->memory->get_layout() != layout) {
+            it->second->set_memory(get_engine().allocate_memory(layout, false));
```

Comment:
I think memory can be reused in some cases

Reply:
Added to compare and reuse existing allocated variable buffers